R/write_assay_tsv.R

```diff
@@ -22,16 +22,16 @@ write_assay_tsv <- function(output_dir, input_csv, ridx, category = c("bacteria"
   # second category specific - enzyme
   if ("enzyme" %in% category){
 
-    rows_indices <- as.numeric(rownames(input_assay[!is.na(input_assay$target_name), ]))
+    rows_indices <- which(!is.na(input_assay$target_name))
     ASSAY[rows_indices,"TARGET_TYPE"] <- "SINGLE_PROTEIN"
     ASSAY[rows_indices, "TARGET_ORGANISM"] <- input_assay[rows_indices, "Species"]
     ASSAY[rows_indices, "TARGET_TAX_ID"] <- input_assay[rows_indices, "organismTAXID"]
-    unlist(ASSAY[rows_indices, "TARGET_ORGANISM"])
-    ASSAY[rows_indices, "AIDX"] <- paste(rep("Zimmermann", length(rows_indices)), unlist(ASSAY[rows_indices, "ASSAY_STRAIN"]), unlist(ASSAY[rows_indices, "ASSAY_ORGANISM"]), rep("biotransformation",  length(rows_indices)), unlist(ASSAY[rows_indices, "TARGET_ACCESSION"]), sep = "_")
+    #unlist(ASSAY[rows_indices, "TARGET_ORGANISM"])
+    ASSAY[rows_indices, "AIDX"] <- paste("Zimmermann", ASSAY[rows_indices, "ASSAY_STRAIN"], ASSAY[rows_indices, "ASSAY_ORGANISM"], "biotransformation", ASSAY[rows_indices, "TARGET_ACCESSION"], sep = "_")
```

Comment:
I think you don't need the `rep()` calls here, as `paste()` will recycle the shorter arguments,

Similarly I'm not sure it's necessary to call `unlist()`.  Each of those should already be a vector that's extracted from the `data.frame`. (Probably worth checking that though, I'm not 100% sure).

Reply:
I think I recently made the change from rep too. for unlist, I am assuming I used unlist to avoid errors later on. Will check again though